umbraco/Umbraco-CMS
# Clean up and improve project templates

### Prerequisites


### Description
This cleans up and improves the project templates we ship in `Umbraco.Templates` and the build pipeline:
- Moved template files from `build/templates` to root
- Updated `Umbraco.Templates.nuspec` to include all necessary files without having to copy them into a temporary folder (and updated `build.ps1` accordingly)
- Removed duplicate `icon.png` files and use it as NuGet package icon (besides the legacy `iconUrl`)
- Cleaned up project and template files, ensured the telemetry ID is always populated and sorted/named the template parameters in a more clear way:

```
> dotnet new umbraco -h
Umbraco Project (C#)
Author: Umbraco HQ
Description: An empty Umbraco project ready to get started
Options:
  -F|--Framework            The target framework for the project.
                                net5.0    - Target net5.0
                                net6.0    - Target net6.0
                            Default: net5.0

  -v|--version              The version of Umbraco.Cms to add as PackageReference.
                            string - Optional
                            Default: 9.4.0-rc

  --use-https-redirect      Adds code to Startup.cs to redirect HTTP to HTTPS and enables the UseHttps setting.
                            bool - Optional
                            Default: false

  -ce|--SqlCe               Adds the required dependencies to use SQL Compact Edition as database provider (Windows only).
                            bool - Optional
                            Default: false

  --no-restore              If specified, skips the automatic restore of the project on create.
                            bool - Optional
                            Default: false

  --friendly-name           Used to specify the name of the default admin user when using unattended install (stored as plain text).
                            string - Optional

  --email                   Used to specify the email of the default admin user when using unattended install (stored as plain text).
                            string - Optional

  --password                Used to specify the password of the default admin user when using unattended install (stored as plain text).
                            string - Optional

  --connection-string       Database connection string used by Umbraco.
                            string - Optional

  --no-nodes-view-path      Path to a custom view presented with the Umbraco installation contains no published content
                            string - Optional

  -p|--PackageTestSiteName  The name of the package project this should be a test site for.
                            string - Optional
```

![devenv_XDLyN17ImM](https://user-images.githubusercontent.com/1051287/155357869-cfe94613-4499-466a-ab56-c0849a657581.png)


---------------

```
❯ dotnet new umbracopackage -h
Umbraco Package (C#)
Author: Umbraco HQ
Description: An empty Umbraco package/plugin project ready to get started
Options:
  -F|--Framework  The target framework for the project.
                      net5.0    - Target net5.0
                      net6.0    - Target net6.0
                  Default: net5.0

  -v|--version    The version of Umbraco.Cms to add as PackageReference.
                  string - Optional
                  Default: 9.4.0-rc
```

![devenv_yZ0aN182V2](https://user-images.githubusercontent.com/1051287/155358243-005ec0d9-d5da-4e8a-9477-0ad9c262460c.png)

Steps to test:
1. Run the build script (`build\build.ps1`) the check if it still builds correctly and outputs the `Umbraco.Templates` NuGet package
2. Uninstall the existing Umbraco templates: `dotnet new --uninstall Umbraco.Templates`
3. Install the newly built templates: `dotnet new --install build.out/Umbraco.Templates.9.4.0-rc.nupkg`
4. Create new projects using the templates to check if all parameters are correctly set, using `dotnet new umbraco` and `dotnet new umbracopackage`



---
_This item has been added to our backlog [AB#17676](https://dev.azure.com/umbraco/243e7927-03b2-44e2-908f-d4ac7ea5daaa/_workitems/edit/17676)_
